docs/essentials/collections/threadsafe/howtoAddandTakeItems.md

```diff
@@ -1 +1,204 @@
+# How to: Add and Take Items Individually from a BlockingCollection
 
+This example shows how to add and remove items from a [BlockingCollection&lt;T&gt;](http://dotnet.github.io/api/System.Collections.Concurrent.BlockingCollection%601.html) in both a blocking and non-blocking manner. For more information on **BlockingCollection&lt;T&gt;**, see [BlockingCollection Overview](blockingCollectionOverview.md). 
+
+For an example of how to enumerate a **BlockingCollection&lt;T&gt;** until it is empty and no more elements will be added, see [How to: Use ForEach to Remove Items in a BlockingCollection](howtoUseForEachtoRemove.md).
+
+## Example
+
+```csharp
+using System;
+using System.Collections.Concurrent;
+using System.Threading;
+using System.Threading.Tasks;
+
+class Program
+{
+   static void Main()
+   {
+      // Increase or decrease this value as desired.
+      int itemsToAdd = 500;
+
+      // Preserve all the display output for Adds and Takes
+      Console.SetBufferSize(80, (itemsToAdd * 2) + 3);
+
+      // A bounded collection. Increase, decrease, or remove the
+      // maximum capacity argument to see how it impacts behavior.
+      BlockingCollection<int> numbers = new BlockingCollection<int>(50);
+
+
+      // A simple blocking consumer with no cancellation.
+      Task.Run(() =>
+      {
+          int i = -1;
+          while (!numbers.IsCompleted)
+          {
+              try
+              {
+                  i = numbers.Take();
+              }
+              catch (InvalidOperationException)
+              {
+                  Console.WriteLine("Adding was completed!");
+                  break;
+              }
+              Console.WriteLine("Take:{0} ", i);
+
+              // Simulate a slow consumer. This will cause
```

Comment:
Do we know the rationale for simulating a slow consumer?  Looking at the code it seems like you'd see output similar to this:

```
Add: 0, Count=1
Add: 0, Count=1
Add: 0, Count=1
Add: 0, Count=1
```